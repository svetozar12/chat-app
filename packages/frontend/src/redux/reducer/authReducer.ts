import { Action } from "../state";
import { ActionType } from "../state";
import { InitialState } from "../state";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  remember_me: false,
  input: "",
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: "",
};
const reducer = (state: InitialState = initialState, action: Action) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.SAVE_INPUT:
      return { ...state, input: action.payload };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    case ActionType.SIGN_OUT:
      return { state: undefined };
    default:
      return state;
  }
};

export default reducer;
