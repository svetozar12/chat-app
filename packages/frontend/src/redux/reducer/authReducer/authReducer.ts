import { HYDRATE } from "next-redux-wrapper";
import { IAuthState } from "./state";
import { ActionType } from "../../types";
export const initialState: IAuthState = {
  remember_me: false,
  good: "", //good and bad stand for good alert and bad alert
  bad: "",
  loginPrompt: false,
  cookie: {
    username: "",
    id: "",
    token: "",
    refresh_token: "",
  },
};
const reducer = (state = initialState, action: any) => {
  switch (action.type) {
    case HYDRATE:
      return {
        ...state,
        ...action.payload,
      };
    case ActionType.LOGIN_POST:
      return { ...state, state };
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.good };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.bad };
    case ActionType.REMEMBER_ME_CHECK:
      return { ...state, remember_me: action.payload };
    case ActionType.QUICK_LOGIN:
      return { ...state, loginPrompt: action.payload };
    case ActionType.SIGN_IN:
      return { ...state, cookie: action.payload };
    default:
      return state;
  }
};

export default reducer;
export { initialState as AuthState };
