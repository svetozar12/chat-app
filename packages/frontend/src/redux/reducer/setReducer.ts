import { Action, ActionType, InitialState2 } from "../state";
export const initialState = {
  reciever: "",
  chat_inviter: "",
  pageNumber: 2,
  setUserSettings: false,
  setFriendRequest: false,
  toggleCreateGroup: false,
  setMobileNav: false,
  setIsMatch: false,
  setChatSettings: false,
  setModalInvite: false,
  setIsLoggingIn: false,
};
const setReducer = (state: InitialState2 | any | InitialState2[] = initialState, action: Action) => {
  switch (action.type) {
    case ActionType.SET_RECIEVER:
      return { ...state, reciever: action.payload };
    case ActionType.CHAT_INVITER:
      return { ...state, chat_inviter: action.payload };
    case ActionType.INCREMENT_PAGE_NUMBER:
      return { ...state, pageNumber: ++action.payload };
    case ActionType.SET_FRIEND_REQUEST:
      return { ...state, setFriendRequest: action.payload };
    case ActionType.SET_USER_SETTINGS:
      return { ...state, setUserSettings: action.payload };
    case ActionType.TOGGLE_CREATE_GROUP:
      return { ...state, toggleCreateGroup: action.payload };
    case ActionType.SET_MOBILE_NAV:
      return { ...state, setMobileNav: action.payload };
    case ActionType.SET_CHAT_SETTINGS:
      return { ...state, setChatSettings: action.payload };
    case ActionType.SET_IS_MATCH:
      return { ...state, setIsMatch: action.payload };
    case ActionType.SET_MODAL_INVITE:
      return { ...state, setModalInvite: action.payload };
    case ActionType.SET_IS_LOGEDIN:
      return { ...state, setIsLoggingIn: action.payload };
    case ActionType.SIGN_OUT:
      return initialState;
    default:
      return state;
  }
};

export default setReducer;
