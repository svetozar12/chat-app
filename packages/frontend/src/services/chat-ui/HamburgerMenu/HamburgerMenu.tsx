// eslint-disable-next-line import/named
import { Box, BoxProps } from '@chakra-ui/react';
import { css, cx } from '@emotion/css';
import { GiHamburgerMenu } from 'react-icons/gi';
import { IBaseComponent } from '../types';

type Base = IBaseComponent<BoxProps>;
interface IHambugerMenu extends Base {
  toggleHamburger: () => void;
}

function HamburgerMenu(props: IHambugerMenu) {
  const { toggleHamburger, baseProps, chakraProps, style } = props;
  return (
    <Box
      className={cx(
        'flex',
        css`
          position: absolute;
          top: 0;
          left: 0;
          width: 3rem;
          height: 3rem;
          border: 1px solid transparent;
          margin: 1rem;
          &:active {
            background: var(--gradient-first);
            border-radius: 100%;
          }
        `,
      )}
      {...chakraProps}
      {...style}
      {...baseProps}
    >
      <GiHamburgerMenu
        title="hamburger"
        className={css`
          display: inline-block;
          cursor: pointer;
          width: 2rem;
          height: 2rem;
          zindex: -11;
          @media (min-width: 980px) {
            display: none;
          } ;
        `}
        onClick={toggleHamburger}
      />
    </Box>
  );
}

export default HamburgerMenu;
