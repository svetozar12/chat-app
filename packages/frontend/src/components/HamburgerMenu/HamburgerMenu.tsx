import { css, cx } from "@emotion/css";
import { useSelector, useDispatch } from "react-redux";
import { IInitialSet } from "../../services/redux/reducer/setReducer/state";
import { GiHamburgerMenu } from "react-icons/gi";
function HamburgerMenu() {
  const dispatch = useDispatch();
  const state = useSelector((state: { setReducer: IInitialSet }) => state.setReducer);

  return (
    <div
      className={cx(
        "flex",
        css`
          position: absolute;
          top: 0;
          left: 0;
          width: 3rem;
          height: 3rem;
          border: 1px solid transparent;
          margin: 1rem;
          &:active {
            background: var(--gradient-first);
            border-radius: 100%;
          }
        `,
      )}
    >
      <GiHamburgerMenu
        title="hamburger"
        className={css`
          display: inline-block;
          cursor: pointer;
          width: 2rem;
          height: 2rem;
          zindex: -11;
          @media (min-width: 980px) {
            display: none;
          } ;
        `}
        onClick={() => dispatch({ type: "SET_MOBILE_NAV", payload: !state.setMobileNav })}
      />
    </div>
  );
}

export default HamburgerMenu;
