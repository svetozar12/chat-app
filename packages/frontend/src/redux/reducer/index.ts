import { combineReducers } from "redux";
import reducer from "./authReducer";
import setReducer from "./setReducer";
import saveInputReducer from "./save_inputReducer";
import messageReducer from "./messageReducer";

const reducers = combineReducers({
  authReducer: reducer,
  setReducer,
  messageReducer,
  saveInputReducer,
});

export default reducers;
