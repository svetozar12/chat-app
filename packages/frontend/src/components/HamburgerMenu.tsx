import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { InitialState2 } from "../redux/state";
import { GiHamburgerMenu } from "react-icons/gi";
function HamburgerMenu() {
  const dispatch = useDispatch();
  const state = useSelector(
    (state: { setReducer: InitialState2 }) => state.setReducer,
  );
  return (
    <GiHamburgerMenu
      className="hamburger"
      onClick={() =>
        dispatch({ type: "SET_MOBILE_NAV", payload: !state.setMobileNav })
      }
    />
  );
}

export default HamburgerMenu;
