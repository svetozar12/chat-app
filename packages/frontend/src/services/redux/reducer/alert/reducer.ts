import { AnyAction } from 'redux';
import { IAlert } from 'services/redux/reducer/alert/state';
import { ActionType } from '../../types';

const initialState: IAlert = {
  good: '', // good and bad stand for good alert and bad alert
  bad: '',
};

const alertReducer = (state = initialState, action: AnyAction) => {
  switch (action.type) {
    case ActionType.LOGIN_POST_ERROR:
      return { ...state, bad: action.payload };
    case ActionType.REGISTER_POST:
      return { ...state, good: action.payload };
    case ActionType.REGISTER_POST_ERROR:
      return { ...state, bad: action.payload };
    default:
      return state;
  }
};

export default alertReducer;
