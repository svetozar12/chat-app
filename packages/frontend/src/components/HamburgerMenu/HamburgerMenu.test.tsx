import HamburgerMenu from "../HamburgerMenu/HamburgerMenu";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { initialState as setState } from "../../redux/reducer/setReducer";
import { AuthState } from "../../redux/reducer/authReducer";
import saveInputReducer from "../../redux/reducer/save_inputReducer";
import { screen, render, cleanup, RenderResult } from "@testing-library/react";
import { ReactTestRendererJSON } from "react-test-renderer";
import "@testing-library/jest-dom";

let component: ReactTestRendererJSON | ReactTestRendererJSON[] | null;
let container: RenderResult;
beforeEach(() => {
  const mockStore = configureStore([]);
  const store = mockStore({
    authReducer: AuthState,
    setReducer: setState,
    saveInputReducer: saveInputReducer,
  });

  component = renderer
    .create(
      <Provider store={store}>
        <HamburgerMenu />
      </Provider>,
    )
    .toJSON();

  container = render(
    <Provider store={store}>
      <HamburgerMenu />
    </Provider>,
  );
});

afterEach(cleanup);

describe("Render connected React-redux page", () => {
  it("should create snapshot for <HamburgerMenu/>", () => {
    expect(component).toMatchSnapshot();
  });

  it("should render <HamburgerMenu/>", () => {
    const renderedComponent = container.getByTitle("hamburger");
    expect(renderedComponent).toBeInTheDocument();
  });
});
