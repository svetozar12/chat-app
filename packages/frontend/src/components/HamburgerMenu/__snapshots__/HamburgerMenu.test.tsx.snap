// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render connected React-redux page should create snapshot for <HamburgerMenu/> 1`] = `
<svg
  className="hide css-1y0ao4k-HamburgerMenu"
  fill="currentColor"
  height="1em"
  onClick={[Function]}
  stroke="currentColor"
  strokeWidth="0"
  style={
    Object {
      "color": undefined,
    }
  }
  viewBox="0 0 512 512"
  width="1em"
  xmlns="http://www.w3.org/2000/svg"
>
  <title>
    hamburger
  </title>
  <path
    d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
  />
</svg>
`;
